src/Helium/CodeGeneration/Iridium/Show.hs

```diff
@@ -224,14 +224,20 @@ instance ShowDeclaration CustomDeclaration where
   showDeclaration (CustomDeclaration kind) = ("custom", ": " ++ showDeclKind kind ++ "\n")
 
 instance ShowDeclaration DataTypeConstructorDeclaration where
-  showDeclaration (DataTypeConstructorDeclaration tp) =
+  showDeclaration (DataTypeConstructorDeclaration tp fs) =
     ( "constructor"
-    , ": { " ++ show tp ++ " }"
+    , ": { " ++ show tp ++ " }" ++ " < " ++ intercalate ", " (map show fs) ++ " > "
     )
 
 instance Show DataTypeConstructor where
   show (DataTypeConstructor name tp) = "@" ++ showId name "" ++ ": " ++ show tp
 
+instance Show Field where
+  show (Field name strict) = showStrict strict ++ showId name ""
+    where
+      showStrict False = ""
+      showStrict True = "!"
```

Comment:
Should we include strictness in Field? As it is already present in type of the constructor